### ISSUE TITLE
`noexcept(false)`-function is labeled as `noexcept`

### ISSUE THREAD

**Issue by e-kwsm** [doxygen/doxygen#9582]
**Describe the bug**
`noexcept` specifier is something like tribool: true, false, or unspecified (which usually implies false).
A function without `noexcept` is not labeled as `noexcept`, but a function with `noexcept` is always labeled as `noexcept`, regardless of its argument.

**Expected behavior**
A function specified as `noexcept` (without argument) or `noexcept(true)` is labeled as `noexcept`.

**Screenshots**
![noexcept](https://user-images.githubusercontent.com/11627708/188624792-d4c03cda-f9fd-46e0-bad7-2b0d54ec6c35.png)

**To Reproduce**
Doxyfile with `EXTRACT_ALL=YES`, and a C++ file:
```cpp
void f() {}                          ///< may throw exception
void g() noexcept {}                 ///< must not throw exception
void h() noexcept(true) {}           ///< must not throw exception
void i() noexcept(false) {}          ///< may throw exception
void j() noexcept(noexcept(h())) {}  ///< must not throw exception
```

[noexcept.zip](https://github.com/doxygen/doxygen/files/9496495/noexcept.zip)

**Version**
1.9.3 installed by pacman; EndeavourOS rolling

**Additional context**
What to do in the case of `noexcept(non_literal_expression)`.

**Comment by albert-github** [comment]
Regarding the constant case `noexcept(false)` I've just pushed a proposed patch, pull request #9583
Regarding the question "What to do in the case of noexcept(non_literal_expression)", doxygen is not a compiler and thus will not interpret the value.

**Comment by e-kwsm** [comment]
I understand the limitation. Thank you for your quick fix.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.6.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
